src/main/java/uk/gov/companieshouse/filinghistory/consumer/apiclient/ResponseHandler.java

```diff
@@ -17,22 +17,23 @@
 public class ResponseHandler {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(NAMESPACE);
-    private static final String API_INFO_RESPONSE_MESSAGE = "Call to API failed, status code: %d. %s";
-    private static final String API_ERROR_RESPONSE_MESSAGE = "Call to API failed, status code: %d.";
-    private static final String URI_VALIDATION_EXCEPTION_MESSAGE = "Failed call to filing history API due to invalid URI";
+    private static final String API_INFO_RESPONSE_MESSAGE = "Call to API failed, status code: %d.";
+    private static final String API_ERROR_RESPONSE_MESSAGE = "Call to API failed, status code: %d. %s";
+    private static final String URI_VALIDATION_EXCEPTION_MESSAGE = "Invalid URI";
 
     public void handle(ApiErrorResponseException ex) {
         final int statusCode = ex.getStatusCode();
         final HttpStatus httpStatus = HttpStatus.valueOf(ex.getStatusCode());
 
         if (HttpStatus.CONFLICT.equals(httpStatus) || HttpStatus.BAD_REQUEST.equals(httpStatus)) {
-            LOGGER.error(API_ERROR_RESPONSE_MESSAGE.formatted(statusCode), ex,
+            LOGGER.error(API_ERROR_RESPONSE_MESSAGE.formatted(statusCode, Arrays.toString(ex.getStackTrace())), ex,
                     DataMapHolder.getLogMap());
-            throw new NonRetryableException(API_ERROR_RESPONSE_MESSAGE.formatted(statusCode), ex);
-        }  else {
-            LOGGER.info(API_INFO_RESPONSE_MESSAGE.formatted(statusCode, Arrays.toString(ex.getStackTrace())),
+            throw new NonRetryableException(API_ERROR_RESPONSE_MESSAGE.formatted(
+                    statusCode, Arrays.toString(ex.getStackTrace())), ex);
+        } else {
+            LOGGER.info(API_INFO_RESPONSE_MESSAGE.formatted(statusCode),
```

Comment:
Why was the stack trace removed from here?

Reply:
I didn't think we would want the stack trace to be logged a) on an info log level, and b) on something that is going to be retried - this could result in multiple stack traces being logged.

Reply:
Do we want the stack trace printed on an info log and in a block of code that might get retried?

Reply:
Reverting after discussion - stack trace will be useful for FESS in info logs.